### PR TITLE
Add OPA Engine Authorizer

### PR DESCRIPTION
**Pull request by namlh-teko**
## Checklist

- [ ] I have read the [contributing guidelines](CONTRIBUTING.md)
- [ ] I have read the [security policy](SECURITY.md)
- [ ] I confirm that this pull request does not address a security
      vulnerability. If this pull request addresses a security vulnerability, I
      confirm that I got green light (please contact
      [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push
      the changes.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation within the code base (if appropriate)
- [ ] I have documented my changes in the
      [developer guide](https://github.com/ory/docs) (if appropriate)

### CODE REVIEW

File: pipeline/authz/keto_engine_acp_ory.go [hunk 1]
```diff
@@ -147,7 +147,7 @@ func (a *AuthorizerKetoEngineACPORY) Authorize(r *http.Request, session *authn.A
 	if res.StatusCode == http.StatusForbidden {
 		return errors.WithStack(helper.ErrForbidden)
 	} else if res.StatusCode != http.StatusOK {
-		return errors.Errorf("expected status code %d but got %d", http.StatusOK, res.StatusCode)
+		return errors.Errorf("Expected status code %d but got %d", http.StatusOK, res.StatusCode)
```
Comment (aeneasr) [inline pipeline/authz/keto_engine_acp_ory.go]: Errors should be lowercase, please revert this change

File: pipeline/authz/opa_engine.go [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+	"github.com/ory/oathkeeper/driver/configuration"
+	"github.com/ory/oathkeeper/helper"
+	"github.com/ory/oathkeeper/pipeline"
+	"github.com/ory/oathkeeper/pipeline/authn"
+
+	"github.com/ory/x/httpx"
+)
+
+type AuthorizerOPAEngineConfiguration struct {
+	RequiredAction   string `json:"required_action" valid:",required"`
```
Comment (aeneasr) [inline pipeline/authz/opa_engine.go]: This assumes the input parameters for OPA to have a specific format. That is an antipattern because it is specific to your use case. Instead, we should use go templates to generate the input values, as it is done with the [header](https://github.com/ory/oathkeeper/blob/master/pipeline/mutate/mutator_header.go) mutator.

**Comment by namlh-teko** [comment]
Thank you for your kind response.

Actually, it's my experiment code to integrate Oathkeeper with OPA Engine. I must open this PR by mistake.
I'm sorry for taking your time in reviewing this PR. Thank you again for your kind response.

If Oathkeeper plans to have an OPA Engine Authorizer, you can reopen this PR, and I'll willing to help.
For now, I'll close this PR.

Thank you.
